### DISCUSSION TITLE
I use coco128 to train custom data but failed  ，program not responding

### DISCUSSION THREAD

**Discussion by snow-bean** [ultralytics/yolov5#7583] [Q&A] [answered]
AutoAnchor: 4.27 anchors/target, 0.994 Best Possible Recall (BPR). Current anchors are a good fit to dataset ✅
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to runs/train/exp8
Starting training for 3 epochs...

     Epoch   gpu_mem       box       obj       cls    labels  img_size
  0%|          | 0/8 [00:00<?, ?it/s]   
<img width="1282" alt="image" src="https://user-images.githubusercontent.com/71054627/165214555-2208c26b-5919-4acb-b6fe-0d6f2123fc8b.png">

After the program runs here, nothing happens。It seems stop
anybody can help?thank you very much

**Comment by glenn-jocher** [comment, marked as answer]
@snow-bean you might just need to restart your computer or restart your environment.

**Reply by snow-bean** [reply]
thank you so much, I am surprised to receive your answer。My first time asking a question on github, thank you very much for your answer. I'll try it right away.

**Reply by snow-bean** [reply]
hello sir maybe I need your help again , so sorry to disturb you. i restart my computer again , but still the same situation ,stop at there     :
Epoch   gpu_mem       box       obj       cls    labels  img_size
  0%|          | 0/8 [00:00<?, ?it/s]    

all I do as below:
(base) hld@hlddeMacBook-Air yolov5 % cd /Users/hld/Downloads/yolov5-master/yolo5qstart/yolov5
(base) hld@hlddeMacBook-Air yolov5 % conda activate pytorch37
(pytorch37) hld@hlddeMacBook-Air yolov5 % python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt

**Reply by snow-bean** [reply]
I want to use yolov5 to recognize some cell pictures, I am learning for the first time, so I try the tutorial on the official website for training， use coco128 train custom data. I only try this  python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt ,and I do not change anything, where do I did wrong ?

**Reply by snow-bean** [reply]
> @snow-bean you might just need to restart your computer or restart your environment.

I want to use yolov5 to recognize some cell pictures, I am learning for the first time, so I try the tutorial on the official website for training， use coco128 train custom data. I only try this python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt ,and I do not change anything, where do I did wrong ?

**Reply by glenn-jocher** [reply]
@snow-bean it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual [Python 3.9](https://www.python.org/) environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install` [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) again from scratch.

💡 ProTip! Try one of our verified environments below if you are having trouble with your local environment.


## Requirements

[**Python>=3.7.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

[Models](https://github.com/ultralytics/yolov5/tree/master/models) and [datasets](https://github.com/ultralytics/yolov5/tree/master/data) download automatically from the latest YOLOv5 [release](https://github.com/ultralytics/yolov5/releases) when first requested.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
- **Docker Image**. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by snow-bean** [reply]
maybe it works !!!  thanks for your help. I follow your advice create a new virtual python39 environmental.
without your help , I nearly give up . 
best wishes to you 



Epoch   gpu_mem       box       obj       cls    labels  img_size
  0%|          | 0/8 [00:00<?, ?it/s]                                                                                                                                              0/2        0G   0.04432   0.06962   0.01962       204       640:   0%|          | 0/8 [00:28<?, ?it/s]                                                                      0/2        0G   0.04432   0.06962   0.01962       204       640:  12%|█▎        | 1/8 [00:32<03:49, 32.83s/it]                                                              0/2        0G   0.04388   0.05766   0.01998       157       640:  12%|█▎        | 1/8 [01:00<03:49, 32.83s/it]

**Reply by snow-bean** [reply]
3 epochs completed in 0.235 hours.
Optimizer stripped from runs/train/exp10/weights/last.pt, 14.9MB
Optimizer stripped from runs/train/exp10/weights/best.pt, 14.9MB




Validating runs/train/exp10/weights/best.pt...
Fusing layers... 
Model summary: 213 layers, 7225885 parameters, 0 gradients, 16.5 GFLOPs
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95

> maybe it works !!! thanks for your help. I follow your advice create a new virtual python39 environmental. without your help , I nearly give up . best wishes to you
> 
> Epoch gpu_mem box obj cls labels img_size 0%| | 0/8 [00:00<?, ?it/s] 0/2 0G 0.04432 0.06962 0.01962 204 640: 0%| | 0/8 [00:28<?, ?it/s] 0/2 0G 0.04432 0.06962 0.01962 204 640: 12%|█▎ | 1/8 [00:32<03:49, 32.83s/it] 0/2 0G 0.04388 0.05766 0.01998 157 640: 12%|█▎ | 1/8 [01:00<03:49, 32.83s/it]

**Reply by glenn-jocher** [reply]
Great!
